device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/Application.java

```diff
@@ -0,0 +1,168 @@
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static samples.com.microsoft.azure.sdk.iot.DeviceTwinHelper.*;
+import static samples.com.microsoft.azure.sdk.iot.MethodHelper.DeviceMethodStatusCallBack;
+import static samples.com.microsoft.azure.sdk.iot.MethodHelper.SampleDeviceMethodCallback;
+import static samples.com.microsoft.azure.sdk.iot.ReceiveMessageHelper.MessageCallback;
+import static samples.com.microsoft.azure.sdk.iot.TelemetryHelper.EventCallback;
+import static samples.com.microsoft.azure.sdk.iot.TelemetryHelper.composeMessage;
+
+@Slf4j
+public class Application {
+    private static final String deviceConnectionString = System.getenv("DEVICE_CONNECTION_STRING");
+    private static final IotHubClientProtocol protocol = IotHubClientProtocol.AMQPS; // Can be configured to use any protocol from  HTTPS, AMQPS, MQTT, AMQPS_WS, MQTT_WS
+    private static final int DEVICE_OPERATION_TIMEOUT_IN_MINUTES = 2;
+    private static final int numRequests = 20;
+    private static final int MAX_APPLICATION_RUN_DURATION_IN_MINUTES = 5;
+    private static final int SLEEP_DURATION_IN_SECS = 60;
+    private static AtomicBoolean stopRunning = new AtomicBoolean(false);
+
+    static boolean reconnectIndefinitely = true;
+    static List<String> failedMessageListOnClose = new ArrayList<>(); // List of messages that failed on close
+    private static CountDownLatch countDownLatch = new CountDownLatch(2);
+    private static DeviceClientManager deviceClientManager;
+
+    /**
+     * Sends a number of messages to an IoT. Default protocol is to use AMQP transport.
+     *
+     */
+    public static void main(String[] args)
+            throws IOException, URISyntaxException, InterruptedException {
+        log.debug("Starting the sample...");
+        log.debug("Using communication protocol: {}", protocol.name());
+
+        DeviceClient client = new DeviceClient(deviceConnectionString, protocol);
+        log.debug("Successfully created an IoT Hub client.");
+
+        deviceClientManager = new DeviceClientManager(client);
+        deviceClientManager.setOperationTimeout(DEVICE_OPERATION_TIMEOUT_IN_MINUTES);
+
+        deviceClientManager.connect();
+        log.debug("Opened connection to IoT Hub.");
+
+        log.debug("Setting C2D message handler...");
+        deviceClientManager.setMessageCallback(new MessageCallback(), new Object());
+
+        log.debug("Setting method handler...");
+        deviceClientManager.subscribeToDeviceMethod(new SampleDeviceMethodCallback(), null, new DeviceMethodStatusCallBack(), null);
+
+        log.debug("Start device Twin and get remaining properties...");
+        // Properties already set in the Service will shows up in the generic onProperty callback, with value and version.
+        Succeed.set(false);
+        client.startDeviceTwin(new DeviceTwinStatusCallBack(), null, new onProperty(), null);
```

Comment:
create a sync function to start twin (instead of looping and waiting on status change in the callback)